<launch>
<!-- Launch file for Livox AVIA LiDAR -->
	<arg name="rviz" default="true" />

	<rosparam command="load" file="$(find fast_lio)/config/avia.yaml" />

    <param name="use_tracking_camera" type="bool" value="1"/>
    <param name="perform_only_localization" type="bool" value="0"/>
	<param name="feature_extract_enable" type="bool" value="0"/>
	<param name="point_filter_num" type="int" value="3"/>
	<param name="max_iteration" type="int" value="3" />
	<param name="filter_size_surf" type="double" value="0.3" />
	<param name="filter_size_map" type="double" value="0.3" />
	<param name="cube_side_length" type="double" value="1000" />
	<param name="runtime_pos_log_enable" type="bool" value="0" />
    <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" />

    <node pkg="tf2_ros" type="static_transform_publisher" name="link_lidar_broadcaster" args="-1.8733e-01 0.016 2.0341e-01 0.0109029 -0.1205037 0.0039972 0.992645 camera_pose_frame LiDAR_frame" />

	<group if="$(arg rviz)">
	<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/loam_livox.rviz" />
	</group>

</launch>
